### ISSUE TITLE
Cannot seem to mock explicitly void methods

### ISSUE THREAD

**Issue by adamcameron** [kahlan/kahlan#266]
Hi there's a gist with the repo case in it here: https://gist.github.com/adamcameron/cdec67d80c1ba55cfda96baa65ba192a

The result of this is:

```
src directory  : /var/data/www/users/adam.cameron/kahlan/src
spec directory : /var/data/www/users/adam.cameron/kahlan/spec

.E                                                                  2 / 2 (100%)

mocked tests
  ✖ it does not like type-checked void methods
    an uncaught exception has been thrown in `/tmp/kahlan/var/data/www/users/adam.cameron/kahlan/src/service/Dependency.php` line 14

    message:`Exception` Code(0) with message "isExplicitVoidMethod was not mocked"

      [NA] - /tmp/kahlan/var/data/www/users/adam.cameron/kahlan/src/service/Dependency.php, line  to 14
      me\adamcameron\kahlan\service\Dependency::isExplicitVoidMethod() - /tmp/kahlan/var/data/www/users/adam.cameron/kahlan/src/service/Service.php, line 21
      me\adamcameron\kahlan\service\Service::callExplicitVoidMethod() - /var/data/www/users/adam.cameron/kahlan/spec/unit/service/serviceSpec.php, line 36
      Kahlan\Specification::_execute() - /var/data/www/users/adam.cameron/kahlan/vendor/kahlan/kahlan/src/Specification.php, line 103
      Kahlan\Specification::_process() -
[etc]
```

The only difference between the methods is the type-check on the return type:

```
public function isImplicitVoidMethod()
{
    throw new \Exception("isImplicitVoidMethod was not mocked");
}

public function isExplicitVoidMethod() : void
{
    throw new \Exception("isExplicitVoidMethod was not mocked");
}
```

This is a distilled repro case from a replicable real-life problem, and have reproed on PHP 7.1.6 on both Windows and CentOS.

Let me know if:
a) I'm doing something wrong;
b) you'd like more information.

Cheers.

**Comment by jails** [comment]
Thank you for the detailed reproducible use case !
Indeed `void` methods didn't bail out as expected.
It's now fixed in the [dev branch](https://github.com/kahlan/kahlan/commit/de551bc7c11f8faffe156196f8485438817271d8)
Once confirmed, I'll tag a new release.

Ps:
caches must be invalidated (`rm -rf /tmp/kahlan` on linux or use `touch` on `Dependency.php` to update modified timestamp of the file)

**Comment by samsonasik** [comment]
I confirm in learn-kahlan repo that dev branch fixed the issue. https://github.com/sitrunlab/learn-kahlan/commit/961a2265d185e2355cdba2b619b306ed51fc5557

**Comment by adamcameron** [comment]
Confirmed fixed for me on Win & CentOS:

```
src directory  : /var/data/www/users/adam.cameron/kahlan/src
spec directory : /var/data/www/users/adam.cameron/kahlan/spec

..                                                                  2 / 2 (100%)



Expectations   : 2 Executed
Specifications : 0 Pending, 0 Excluded, 0 Skipped

Passed 2 of 2 PASS in 0.048 seconds (using 1Mo)
```
Nice work team!

**Comment by jails** [comment]
Fixed [in 3.1.17](https://github.com/kahlan/kahlan/commit/7bc58d8561d467c2913e668638f9079f4a78aef3), thanks everybody ! :+1:
